File: CSharp/Boardgames/Shared/Message.cs
namespace Shared;

public class Message {
    public Head Head { get; }
    public object Body { get; }

    public Message(Head head, object body) {
        Head = head;
        Body = body;
    }

    public Message(object body, string jwt) {
        Head = new Head(body.GetType().Name, jwt, 0.0);
        Body = body;
    }
}

File: Java/boardgames/logic/src/main/java/boardgames/logic/messages/Message.java
package boardgames.logic.messages;

import static boardgames.logic.messages.Messages.*;

public record Message(Head head, Object body) {
    public Message(Object body, double serverMillis) {
        this(new Head(body.getClass().getSimpleName(), "", serverMillis), body);
    }

    public Message(Object body) {
        this(body, 0.0);
    }
}
